test-suite/CMakeLists.txt

```diff
@@ -236,16 +236,23 @@ if(DJINNI_WITH_PYTHON)
 
   # Prepare Python 3 environment
   find_package(PythonInterp "3" REQUIRED)
-  string(JOIN ":" PYTHONPATH
+  if (WIN32) # TODO, obviously that is not optimal
```

Comment:
Why is this not optimal? Because of the code duplication inside the if/else?

Maybe this is more optimal?

```cmake

if (WIN32)
    set(PYTHONPATH_SEPARATOR ":")
else()
    set(PYTHONPATH_SEPARATOR ";")
endif()

string(JOIN ${PYTHONPATH_SEPARATOR} PYTHONPATH
     "${CMAKE_SOURCE_DIR}/djinni/py"
      "${CMAKE_CURRENT_BINARY_DIR}/generated-src/python"
      $ENV{PYTHONPATH})

```

Reply:
exactly :-)

Reply:
change the ";" against a generator expression so cmake does not escape it to a list
but might do that improvement also, will see